pilot/pkg/features/telemetry.go

```diff
@@ -76,4 +76,7 @@ var (
 
 	EnableControllerQueueMetrics = env.Register("ISTIO_ENABLE_CONTROLLER_QUEUE_METRICS", false,
 		"If enabled, publishes metrics for queue depth, latency and processing times.").Get()
+
+	SpawnUpstreamSpanForGateway = env.Register("PILOT_SPAWN_UPSTREAM_SPAN_FOR_GATEWAY", false,
```

Comment:
Is there a reason we don't always want this?

Reply:
the only reason I think is backward compatibility.

My plan is changed the default value to ture in v1.27 and remove it then.

Reply:
Can we add to compatibility profile then?

Reply:
yes, will add compatibility in v1.27